cmd/stack.go

```diff
@@ -136,6 +136,28 @@ func setupStackCommand() *cobra.Command {
 		},
 	}
 
+	dumpCommand := &cobra.Command{
+		Use:   "dump",
+		Short: "Dump stack data for debug purposes",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			output, err := cmd.Flags().GetString(cobraext.StackDumpOutputFlagName)
+			if err != nil {
+				return cobraext.FlagParsingError(err, cobraext.StackDumpOutputFlagName)
+			}
+
+			err = stack.Dump(stack.DumpOptions{
+				Output: output,
+			})
+			if err != nil {
+				return errors.Wrap(err, "dump failed")
+			}
+
+			cmd.Println("Done")
+			return nil
+		},
+	}
+	dumpCommand.Flags().StringP(cobraext.StackDumpOutputFlagName, "", "elastic-stack-dump", cobraext.StackDumpOutputFlagDescription)
```

Comment:
How about changing the default value to `build/elastic-stack-dump`? It's consistent with `build/test-results`, which have a similar feel IMO.

Reply:
I did it on purpose, because the `elastic-package stack` is executed in the global scope, so it may happen there won't be the `build` directory. Do you agree with this behavior?

Reply:
Hmm, I see your point. 

Perhaps I should explain my concern: I was rather hoping we would have a single top-level directory where we can collect all runtime artifacts rather than multiple top-level directories, increasing the level of "pollution" in a project using `elastic-package`. 

> ... so it may happen there won't be the `build` directory. 

In this case couldn't we create the `build` directory if it doesn't exist?

That said, I don't feel too strongly about this either way. We can always revisit the behavior and change if it becomes a problem later.